src/commands/farm.rs

```diff
@@ -135,31 +136,60 @@ async fn start_farming(farming_args: FarmingArgs) -> Result<(Farmer, Node)> {
 
 #[instrument]
 async fn prepare_farming() -> Result<FarmingArgs> {
-    let config_args = parse_config()?;
-
-    let node_name = config_args.node_config_args.name;
-    let chain = match config_args.node_config_args.chain.as_str() {
+    let config = parse_config()?;
+
+    let node_config = config.node;
+    let NodeConfig {
+        chain,
+        execution: _,
+        blocks_pruning,
+        state_pruning,
+        validator,
+        name,
+        port,
+        unsafe_ws_external,
+    } = node_config;
+
+    let chain = match chain.as_str() {
         "dev" => chain_spec::dev_config().unwrap(),
         _ => unreachable!("there are no other valid chain-specs at the moment"),
     };
-    let is_validator = config_args.node_config_args.validator;
-    let role = match is_validator {
+    let role = match validator {
         true => subspace_sdk::node::Role::Authority,
         false => subspace_sdk::node::Role::Full,
     };
+    let rpc_method = match unsafe_ws_external {
+        true => RpcMethods::Unsafe,
+        false => RpcMethods::Auto,
+    };
+    let state_pruning = Some(PruningMode::Constrained(Constraints {
+        max_blocks: Some(state_pruning),
+        max_mem: None,
+    }));
+    let blocks_pruning = BlocksPruning::Some(blocks_pruning);
+    let listen_on = vec![port
+        .parse()
+        .map_err(|_| Report::msg("port in the config file is incorrect"))?];
```

Comment:
`port` here is multiaddr right? So maybe it is not a good name for it.

Also, maybe you could ask users for a vector of multiaddresses.

Reply:
Exactly. I was thinking of changing the name. But `Multiaddress` is not a user-friendly name I guess. It would be confusing. 
I agree on `port` is misleading, but couldn't find a better candidate tbh.

Reply:
you can just have it like that:
```toml
listen_on = ["/ip4/127.0.0.1/0"]
```

Reply:
will this value be valid for each chain spec and each setting? If we may not want this hardcoded value to be present in any case, I don't want to enable it by default